### PR TITLE
Serve only files when running `lhci collect`

### PR DESCRIPTION
**Pull request by ojizero**
Fixes the behaviour where the FallbackServer attempts to serve folders that
happen to end with `.html` causing errors. Filters scanned directory
for only files in it before executing any of the logic there.

Fixes: #444

### CODE REVIEW

**Review by patrickhulce — COMMENTED** [review]
awesome thank you @ojizero! would you be able to add this to one of our autorun test cases? 

I'm thinking it's fine if you just add a file `folder.html/invisible.html` to https://github.com/GoogleChrome/lighthouse-ci/tree/master/packages/cli/test/fixtures/autorun-static-dir/build  and make sure we don't pick it up in the existing test, no need to write a new one

**Comment by ojizero** [comment]
@patrickhulce hmmmm 🤔 just thinking out loud here but isn't the condition `if (folder.includes('.')) continue;` kind of an overkill? maybe the URL does have `.` in it and has other things living under it? aren't those valid pages to check on?

shouldn't the check be only for hidden files? i.e. `something.other/somepage.html` should be a valid page that gets audited? maybe this can be treated in another PR but I'm just thinking out loud here since I think the suggested `folder.html/invisible.html` should in fact be visible given that the depth being checked is `2` 🤔

**Comment by patrickhulce** [comment]
> isn't the condition if (folder.includes('.')) continue; kind of an overkill? maybe the URL does have . in it and has other things living under it? aren't those valid pages to check on?

Maybe, but I don't think we should change it here for two reasons.

1) This is just for automatic defaults, it's not restrictive of what you can do. A folder with a dot is at the very least unusual and I'm OK with those URLs needing explicit opt-in.
2) Any change that automatically starts including *more* files here would need to be in a major breaking version per [version policy](https://github.com/GoogleChrome/lighthouse-ci/blob/master/docs/version-policy.md).

**Comment by ojizero** [comment]
very true yea changing this would be a breaking change and may introduce unexpected behaviours ! I'll add the new fixture for the test as it is now 👍

File: packages/cli/test/fixtures/autorun-static-dir/build/folder.html/invesible.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
```
Comment (patrickhulce) [inline packages/cli/test/fixtures/autorun-static-dir/build/folder.html/invesible.html:1]: can you rename this file `invisible.html`? :) just spelling typo

**Review by patrickhulce — APPROVED** [review]
implementation looks great to me, thanks @ojizero! 🎉

File: packages/cli/test/fixtures/autorun-static-dir/build/folder.html/invesible.html (hunk 1 above)
Comment (ojizero) [reply packages/cli/test/fixtures/autorun-static-dir/build/folder.html/invesible.html:1]: if I got a dollar for every typo I make 💃

**Pull request merged**